### ISSUE TITLE
DMSf Installation error on "bundle install"

### ISSUE THREAD

**Issue by vergelli** [danmunn/redmine_dmsf#1261]
Hello, I'm having an issue when installing the newest version of DMSf, however, the problem itself might not be related to DMSf, so if you consider that this does not belong here I will understand it.  _Sorry in advance_. 
But I thought that maybe you would be familiar with this error and know a practical way to tackle it. 

On point 4.:
"4. Install dependencies: bundle install"

After running: `bundle install`

I recieve:

```
The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x64-mingw32, x86-mswin32. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x64-mingw32 x86-mswin32`.
Fetching gem metadata from https://rubygems.org/.............
Fetching gem metadata from https://rubygems.org/.............
Fetching gem metadata from https://rubygems.org/.
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Using rake 13.0.3
Using concurrent-ruby 1.1.8
.
.
```
**A lot of dependencies later ..** 
```
.
.
Fetching redmine_extensions 0.3.11
Installing active_record_union 1.3.0
Installing redmine_extensions 0.3.11
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /usr/local/bundle/gems/ox-2.14.4/ext/ox
/usr/local/bin/ruby -I /usr/local/lib/ruby/2.6.0 -r ./siteconf20210603-65-n73kag.rb extconf.rb
>>>>> Creating Makefile for ruby version 2.6.7 on x86_64-linux <<<<<
checking for rb_time_timespec()... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
        --with-opt-dir
        --without-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=.
        --curdir
        --ruby=/usr/local/bin/$(RUBY_BASE_NAME)
        --with-ox-dir
        --without-ox-dir
        --with-ox-include
        --without-ox-include=${ox-dir}/include
        --with-ox-lib
        --without-ox-lib=${ox-dir}/lib
/usr/local/lib/ruby/2.6.0/mkmf.rb:467:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:552:in `try_link0'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:570:in `try_link'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:782:in `try_func'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:1069:in `block in have_func'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:959:in `block in checking_for'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:361:in `block (2 levels) in postpone'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:331:in `open'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:361:in `block in postpone'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:331:in `open'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:357:in `postpone'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:958:in `checking_for'
        from /usr/local/lib/ruby/2.6.0/mkmf.rb:1068:in `have_func'
        from extconf.rb:35:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /usr/local/bundle/extensions/x86_64-linux/2.6.0/ox-2.14.4/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /usr/local/bundle/gems/ox-2.14.4 for inspection.
Results logged to /usr/local/bundle/extensions/x86_64-linux/2.6.0/ox-2.14.4/gem_make.out

An error occurred while installing ox (2.14.4), and Bundler cannot continue.
Make sure that `gem install ox -v '2.14.4' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  ox
```

Seems there is a problem with the `ox` Gem.
To be honest, I'm not an Expert with Ruby or Rails, and before starting to touch, and try the recommendations that are being shown there, (with the risk I break everything  >.<" ), first I'd like to know if you know a practical way to solve this issue.

Sorry if this is Offtopic.

Regards

**Comment by picman** [comment]
It's very probably a missing library in your system. What is in _/usr/local/bundle/extensions/x86_64-linux/2.6.0/ox-2.14.4/mkmf.log_?

**Comment by vergelli** [comment]
This is what I've got:
```

"gcc -o conftest -I/usr/local/include/ruby-2.6.0/x86_64-linux -I/usr/local/include/ruby-2.6.0/ruby/backward -I/usr/local/include/ruby-2.6.0 -I.    -DRUBY_TYPE=ruby -DRUBY_RUBY -DRUBY_VERSION=2.6.7 -DRUBY_VERSION_MAJOR=2 -DRUBY_VERSION_MINOR=6 -DRUBY_VERSION_MICRO=7 -Wall -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wduplicated-cond -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wrestrict -Wwrite-strings -Wimplicit-fallthrough=0 -Wmissing-noreturn -Wno-cast-function-type -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-packed-bitfield-compat -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wsuggest-attribute=format -Wsuggest-attribute=noreturn -Wunused-variable  -fPIC conftest.c  -L. -L/usr/local/lib -Wl,-rpath,/usr/local/lib -L. -fstack-protector-strong -rdynamic -Wl,-export-dynamic     -Wl,-rpath,/usr/local/lib -L/usr/local/lib -lruby  -lm   -lc"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

```

**Comment by picman** [comment]
Run `gem install ox -v '2.14.4' --source 'https://rubygems.org/'` and go through all related logs as recommended in the output. There must be an error that has to by fixed.

**Comment by picman** [comment]
Package build-essential installed?

_ox_ package is required by the WebDAV module and is referenced in Gemfile.

**Comment by vergelli** [comment]
Actually, yes, after running _apt-get install -y build-essential_  it worked. It may be good to point out I have this Redmine running on a docker container. 
Thank you 
Regards
